seacrowd/sea_datasets/mywsl2023/mywsl2023.py

```diff
@@ -0,0 +1,172 @@
+
+_LANGUAGES = ["xml"]  # We follow ISO639-3 language code (https://iso639-3.sil.org/code_tables/639/data)
+
+_LICENSE = Licenses.CC_BY_4_0.value
+
+_LOCAL = False
+
+_URLS = {_DATASETNAME: "https://data.mendeley.com/public-files/datasets/zvk55p7ktd/files/7f11b8a0-24e4-45df-af3d-e861f41435ea/file_downloaded"}
+
+_SUPPORTED_TASKS = [Tasks.SIGN_LANGUAGE_RECOGNITION]
+_SUPPORTED_SCHEMA_STRINGS = [f"seacrowd_{str(TASK_TO_SCHEMA[task]).lower()}" for task in _SUPPORTED_TASKS]
+
+_SPLITS = ["train", "test"]
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+class MyWsl2023(datasets.GeneratorBasedBuilder):
+    """This dataset contains pictures of hand gestures corresponding to ten commonly-used Malaysian Sign Language (XML) words."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    subset_id = _DATASETNAME
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{subset_id}_source",
+            version=SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=subset_id,
+        )
+    ]
+
+    seacrowd_schema_config: list[SEACrowdConfig] = []
+
+    for seacrowd_schema in _SUPPORTED_SCHEMA_STRINGS:
+
+        seacrowd_schema_config.append(
+            SEACrowdConfig(
+                name=f"{subset_id}_{seacrowd_schema}",
+                version=SEACROWD_VERSION,
+                description=f"{_DATASETNAME} {seacrowd_schema} schema",
+                schema=f"{seacrowd_schema}",
+                subset_id=subset_id,
+            )
+        )
+
+    BUILDER_CONFIGS.extend(seacrowd_schema_config)
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "image_paths": datasets.Sequence(datasets.Value("string")),
+                    "texts": datasets.Value("string"),
+                }
+            )
+
+        elif self.config.schema == f"seacrowd_{str(TASK_TO_SCHEMA[Tasks.SIGN_LANGUAGE_RECOGNITION]).lower()}":
+            features = schemas.image_text_features(label_names=[])
+
+        else:
+            raise ValueError(f"Invalid config: {self.config.name}")
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        split_generators = []
+
+        path = dl_manager.download_and_extract(_URLS[_DATASETNAME])
+
+        for split in _SPLITS:
+            split_generators.append(
+                datasets.SplitGenerator(
+                    name=split,
```

Comment:
@sabilmakbar our template uses `datasets.Split.{TRAIN,TEST,VALIDATION}` for this `name` field. Do you know if that matters?

Reply:
actually it runs just fine, but let's use the `datasets.Split` method. Thanks for catching this too, @akhdanfadh!